app-doc/heirloom-doctools/heirloom-doctools-160308.ebuild

```diff
@@ -0,0 +1,215 @@
+
+LICENSE="CDDL"
+
+SLOT="0"
+
+KEYWORDS="~amd64 ~x86"
+
+IUSE="cxx"
+
+RDEPEND="!sys-apps/groff"
+DEPEND="sys-devel/flex
+	virtual/yacc"
+
+src_prepare() {
+	# Make sure that C++ code is built with CXXFLAGS and not CFLAGS.
+	find . -name Makefile.mk -exec \
+		sed -i \
+			-e '/(CCC)/s:CFLAGS:CXXFLAGS:' \
+		{} +
+
+	# mpm uses C++, we'll build it explicitly if we really want to
+	sed -i -e 's:mpm:$(MPM):' makefile
+
+	# Monkeypatching dependencies to avoid parallel make failure
+	echo "picl.o: picl.c y.tab.h" >> pic/Makefile.mk
+
+  # Fixing compilation issues
+	sed -i -e 's:e.h:e.h y.tab.h:' eqn/eqn.d/Makefile.mk
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/eqn.d/Makefile.mk
+
+  sed -i -e 's:e.h:e.h y.tab.h:' eqn/neqn.d/Makefile.mk
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/neqn.d/Makefile.mk
+
+	# Move some stuff around to better suit our filesystem layout
+  sed -i -e '/INSTALL.*grap\.defines/s:$(LIBDIR):/usr/share/heirloom/doctools:' \
+		grap/Makefile.mk || die
+	sed -i -e '/GRAPDEFINES/s:LIBDIR:"/usr/share/heirloom/doctools":' \
+		grap/main.c || die
+
+	sed -i -e '/INSTALL.*eign/s:$(LIBDIR):/usr/share/heirloom/doctools:' \
+		ptx/Makefile.mk || die
+	sed -i -e '/\/eign/s:\(LIB\|REF\)DIR:"/usr/share/heirloom/doctools":' \
+		ptx/ptx.c refer/mkey3.c || die
+
+	sed -i -e 's:$(LIBDIR)/vgrindefs:/usr/share/heirloom/doctools/vgrindefs:' \
+		vgrind/Makefile.mk || die
+	sed -i -e '/\/vgrindefs/s:LIBDIR:"/usr/share/heirloom/doctools":' \
+		vgrind/vfontedpr.c || die
+
+	sed -i -e 's:$(REFDIR)/papers:/usr/share/heirloom/reftools/papers:g' \
+		refer/Makefile.mk || die
+	sed -i -e '/\/papers\/Ind/s:REFDIR:"/usr/share/heirloom/reftools":' \
+		refer/refer1.c || die
+
+	# Correct paths for the installed man pages, just to be clean
+	find . \( -name '*.1' -or -name '*.1b' -or -name '*.7' -or -name '*.7b' \) -exec \
+		sed -i \
+			-e "s:/usr/ucblib/grap.defines:/usr/share/heirloom/doctools/grap.defines:" \
+			-e "s:/usr/ucblib/vgrindefs:/usr/share/heirloom/doctools/vgrindefs:" \
+			-e "s:/usr/ucblib/vfontedpr:/usr/libexec/heirloom/doctools/vfontedpr:" \
+			-e "s:/usr/ucblib/tmac/vgrind:/usr/share/heirloom/tmac/vgrind:" \
+			-e "s:/usr/ucblib/eign:/usr/share/heirloom/doctools/eign:" \
+			-e "s:/usr/ucb/:/usr/bin/:g" \
+			-e "s:/usr/ucblib/doctools:/usr/share/heirloom/doctools:g" \
+			-e "s:/usr/ucblib/reftools/papers:/usr/share/heirloom/reftools/papers:" \
+			-e "s:/usr/ucblib/reftools:/usr/libexec/heirloom/reftools:g" \
+		{} +
+
+	eapply_user
+}
+
+src_configure() {
+	append-cppflags -D_GNU_SOURCE
+
+	sed \
+		-e "s:@CFLAGS@:${CFLAGS}:" \
+		-e "s:@CXXFLAGS@:${CXXFLAGS}:" \
+		-e "s:@CPPFLAGS@:${CPPFLAGS}:" \
+		-e "s:@LDFLAGS@:${LDFLAGS}:" \
+		-e "s:@CC@:$(tc-getCC):" \
+		-e "s:@CXX@:$(tc-getCXX):" \
+		-e "s:@RANLIB@:$(tc-getRANLIB):" \
+		-e "s:@libdir@:$(get_libdir):" \
+		"${FILESDIR}"/${PV}.config \
+		> "${S}"/mk.config
+}
+
+src_compile() {
+	emake $(use cxx && echo MPM=mpm) \
+  BINDIR="/usr/bin" \
```

Comment:
Misindent. Always use tab-indent, and increase indent of 2+ lines for readability.

I don't really get this. First you sed all of the paths out, then you pass them to Makefile…